docs/ledger.md

```diff
@@ -0,0 +1,75 @@
+# Ledger
+
+This section describes the design of the distributed public ledger abstraction
+described in the core module.
+
+The system is built around three abstractions that offer an API to clients so
+that one can propagate transactions, create and validate that it will be
+accepted, and finally wait for a confirmation that it is included, or rejected,
+by the public ledger.
+
+A transaction is what triggers a change in the ledger set of values, where it is
+left to the ledger implementation to define how values are stored and
+identified. The transaction is created with a targeted execution environment
+alongside the input parameters.
+
+The first abstraction which allows one to propagate transactions is defined as a
+pool that can work independetly from the other abstractions. In fact, you could
+run a distributed system that only shares transactions.
+
+The second is a service that accepts a transaction and can return if it will be
+included in the next block, or if it will be rejected. Another aspect defines
+the validity of the transaction so that it cannot be reused in a _replay attack_
+for instance.
+
+Finally, the third abstraction will collect the transactions from the pool and
+create a chain of blocks, block after block according to a consensus algorithm.
+One can register to the service to receive notifications when a new block is
+created and learn about the transactions included in the block, and which of
+them have been accepted.
+
+## Transaction Pool
+
+Each participant of a distributed ledger needs to collect a bunch of
+transactions when it is trying to create a block. The pool is there to provide
+that service, so that clients can send their transactions to one of the pool of
+the distributed system while the ordering services wait for enough transactions
+to be discovered to start a new block.
+
+The purpose is to offer a single entry point for the client and then the system
+will take care of spreading the transactions so that any reachable member will
+at some point learn about it.
```

Comment:
And if you don't trust the system, better contact multiple pools on different nodes it one wants to be sure the tx won't be censored. Right ?

Reply:
Yes that's a good point